includes/razorpay-webhook.php

```
@@ -0,0 +1,106 @@
+<?php
+
+require_once __DIR__.'/../razorpay-payments.php';
+
+require_once __DIR__.'/../razorpay-sdk/Razorpay.php';
+use Razorpay\Api\Api;
+
+class RZP_Webhook
+{
+    protected $razorpay;
+    protected $api;
+
+    function __construct()
+    {
+        $this->razorpay = new WC_Razorpay();
+
+        $this->api = $this->razorpay->getRazorpayApiInstance();
+    }
+
+    public function process()
+    {
+        $post = file_get_contents('php://input');
+
+        $data = json_decode($post, true);
+
+        if ($this->razorpay->enable_webhook === 'yes' && empty($data['event']) === false)
+        {
+            if ((isset($_SERVER['HTTP_X_RAZORPAY_SIGNATURE']) === true))
+            {
+                $razorpayWebhookSecret = $this->razorpay->webhook_secret;
+
+                //
+                // If the webhook secret isn't set on wordpress, return
+                //
+                if (empty($razorpayWebhookSecret) === true)
+                {
+                    return;
+                }
+
+                try
+                {
+                    $this->api->utility->verifyWebhookSignature($post,
+                                                                $_SERVER['HTTP_X_RAZORPAY_SIGNATURE'],
+                                                                $razorpayWebhookSecret);
+                }
+                catch (Exception $e)
```

Don't catch all exceptions and we need to log here.